com.unity.hlod/Editor/Utils/WorkingCollider.cs

```diff
@@ -1,106 +1,107 @@

-                parameters.Convex = meshCollider.convex;

-            }

-            else if(collider is SphereCollider sphereCollider)

-            {

-                parameters.CenterX = sphereCollider.center.x;

-                parameters.CenterY = sphereCollider.center.y;

-                parameters.CenterZ = sphereCollider.center.z;

-                parameters.Radius = sphereCollider.radius;

-            }

-            else if(collider is CapsuleCollider capsuleCollider)

-            {

-   

-                parameters.CenterX = capsuleCollider.center.x;

-                parameters.CenterY = capsuleCollider.center.y;

-                parameters.CenterZ = capsuleCollider.center.z;

-                parameters.Radius = capsuleCollider.radius;

-                parameters.Height = capsuleCollider.height;

-                parameters.Direction = capsuleCollider.direction;

-            }

-            else if(collider is TerrainCollider terrainCollider)

-            {

-                parameters.TerrainData = GUIDUtils.ObjectToGUID(terrainCollider.terrainData);

-            }

-            else

-            {

-                Debug.LogError($"{collider.name} collider is not support.");

-            }

-            

-            return wc;

-

-        }

-    }

-    public class WorkingCollider

-    {

-

-        string m_type;

-

-        Vector3 m_position;

-        Quaternion m_rotation;

-        Vector3 m_scale;

-        

-        SerializableDynamicObject m_parameters = new SerializableDynamicObject();

-

-        public string Type

-        {

-            get => m_type;

-            set => m_type = value;

-        }

-

-        public Vector3 Position

-        {

-            get => m_position;

-            set => m_position = value;

-        }

-

-        public Quaternion Rotation

-        {

-            get => m_rotation;

-            set => m_rotation = value;

-        }

-

-        public Vector3 Scale

-        {

-            get => m_scale;

-            set => m_scale = value;

-        }

-

-        public SerializableDynamicObject Parameters => m_parameters;

-    }

+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.HLODSystem.Utils
+{
+    public static class ColliderExtension
+    {
+        public static WorkingCollider ToWorkingCollider(this Collider collider, HLOD hlod)
+        {
+            var hlodWorldToLocal = hlod.transform.worldToLocalMatrix;
+            var colliderLocalToWorld = collider.transform.localToWorldMatrix;
+            var matrix = hlodWorldToLocal * colliderLocalToWorld;
+
+            WorkingCollider wc = new WorkingCollider();
+            dynamic parameters = wc.Parameters;
+
+            wc.Type = collider.GetType().Name;
+            wc.Position = matrix.GetPosition();
+            wc.Rotation = matrix.rotation;
+            wc.Scale = matrix.lossyScale;
```

Comment:
All looks good to me except that because of using 'lossyScale', this cannot handle non-uniformly scaled objects(ex. parent transform is scaled (10, 1, 1), has a child object rotated 30 in y axis). However this isn't something newly introduced and Unity's Physics also doesn't care this, so we can ignore this too. If you confirm whether it's ok, then I'll submit approval immediately :)

Reply:
There is no way to support shear transform. Even the physics engine can't support that. My intent is matching transform with the physics engine. This is works well as my intent.

Reply:
Sounds reasonable! I will submit the approval :) Thanks for the explanation!